### PR TITLE
[cpp worker] support Ray::Get() static method

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?
- Problem
    We just support `Get` to be a member function of object_ref now. The demo like this:
```
  auto obj_ref1 = Ray::Put(100);
  auto res1 = obj_ref1.Get();
  EXPECT_EQ(100, *res1);
```
    And maybe users need a static function(same as python api), like this:
```
  auto obj_ref2 = Ray::Put(200);
  auto res2 = Ray::Get(obj_ref2);
  EXPECT_EQ(200, *res2);
```
- solution
  * Make `Get` method of `Ray` class to be public 

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: cpp/include/ray/api.h [hunk 1]
```diff
@@ -39,6 +39,14 @@ class Ray {
   template <typename T>
   static ObjectRef<T> Put(const T &obj);
 
+  /// Get a single object from the object store.
+  /// This method will be blocked until all the objects are ready.
+  ///
+  /// \param[in] object The object reference which should be got.
+  /// \return shared pointer of the result.
+  template <typename T>
+  static std::shared_ptr<T> Get(const ObjectRef<T> &object);
```
Comment (raulchen) [inline cpp/include/ray/api.h:48]: We can also remove `Get(const std::vector<ObjectID> &ids)` now

Comment (SongGuyang) [reply cpp/include/ray/api.h:48]: We don't support get object for ID?

Comment (raulchen) [reply cpp/include/ray/api.h:48]: Right. Actually we don't need to expose ObjectID to users at all.

File: cpp/include/ray/api.h [hunk 2]
```diff
@@ -39,6 +39,14 @@ class Ray {
   template <typename T>
   static ObjectRef<T> Put(const T &obj);
 
+  /// Get a single object from the object store.
+  /// This method will be blocked until all the objects are ready.
```
Comment (AmeerHajAli) [inline cpp/include/ray/api.h:43]: But you are passing a single object right?
```suggestion
  /// This method will be blocked until the object is ready.
```

File: cpp/include/ray/api.h [hunk 3]
```diff
@@ -39,6 +39,14 @@ class Ray {
...
+  ///
+  /// \param[in] object The object reference which should be got.
```
Comment (AmeerHajAli) [inline cpp/include/ray/api.h:45]:
```suggestion
  /// \param[in] object. The object reference which should be returned.
```

**Review by AmeerHajAli — DISMISSED** [review]

**Comment by AmeerHajAli** [comment]
@SongGuyang , Can you please resolve the comments so we can merge this?

Thanks!

File: cpp/include/ray/api.h (hunk 1 above)
Comment (SongGuyang) [reply cpp/include/ray/api.h:48]: ok

File: cpp/include/ray/api.h (hunk 3 above)
Comment (SongGuyang) [reply cpp/include/ray/api.h:45]: I found no the "." separator in other code, just like core_worker.h?

**Review by raulchen — APPROVED** [review]

**Pull request merged**